### ISSUE TITLE
Replace FGSL calls by direct interface

### ISSUE THREAD

**Issue by krystophny** [itpplasma/NEO-2#18]
See MEPHIT for examples.

**Comment by krystophny** [comment]
Superseded by the migration tracker #88. That tracker plans the FGSL/GSL removal as fortnum leaves (special functions, roots, integration, B-spline) and is the single coordination point. Leaving this open for the historical replacement context; track the work in #88.

**Comment by krystophny** [comment]
*Automated triage pass (Claude), grounded in the repository at `80a6032`. Verify before acting.*

# Refine: keep open, rescope to the one FGSL wrapper that can be deleted today

The maintainer comment of 2026-06-10 on this issue says, in full: *"Superseded by the migration tracker #88. ... Leaving this open for the historical replacement context; track the work in #88."* That is an explicit decision to keep this issue open, so this pass does not close it. What it does propose is to give the issue a body that is actionable on its own instead of a one-line pointer.

## Purpose

Remove one of the four FGSL wrappers from `COMMON/` — `gsl_roots_routines_mod.f90` — without waiting on any fortnum milestone. It is compiled into `libcommon` but has no call sites anywhere in the tree, so deleting it is a pure subtraction: it shrinks the FGSL surface that #88 must migrate, and it is the only part of the FGSL removal that is not blocked.

This rescoping is needed because the issue as originally written is no longer the plan. The literal ask — hand-written `bind(C)` interfaces in the style of MEPHIT — is not what either repo is doing now. NEO-2#88 replaces the wrappers with fortnum leaves, and the MEPHIT pattern this issue points at (`src/mephit_pert.f90:40-49`, the `gsl_sf_bessel_icn_array` interface with `bind(C, name='gsl_sf_bessel_In_array')`, at MEPHIT `fd2c56b`) is itself listed for replacement by `fortnum_special` in the MEPHIT#24 leaf table. A direct `bind(C)` interface would also not remove `find_package(GSL REQUIRED)` (`CMakeLists.txt:108`) or the `GSL::gsl` link (`COMMON/CMakeLists.txt:100`), so it does not address the GPL-2.0+ obligation that motivates the work.

## Current state

Verified on `main` at `80a6032`:

- `COMMON/CMakeLists.txt:4-9` lists four wrappers in `FGSL_USERINTERFACE_SOURCE_FILES`; line 5 is `gsl_roots_routines_mod.f90`. They are compiled into `common` at `COMMON/CMakeLists.txt:92-96`, which links `${LIBFGSL}` and `GSL::gsl` (lines 99-100).
- `cmake/fgsl.cmake` (17 lines) builds FGSL v1.6.0 from `reinh-bader/fgsl` via `ExternalProject_Add`, with `CONFIGURE_COMMAND mkdir m4 && autoreconf -i && ./configure`.
- The only `use` statements naming these modules in the whole tree are:
  - `COMMON/collop_compute.f90:4` — `use gsl_integration_routines_mod`
  - `COMMON/collop_compute.f90:5` — `use gsl_specialfunctions_mod`
  - `COMMON/collop_bspline.f90:3` — `use gsl_bspline_routines_mod`
  - `NEO-2-QL/collop_definitions.f90:3` and `NEO-2-PAR/collop_definitions.f90:3` — `use gsl_integration_routines_mod`
- `rg gsl_roots` over the tree returns exactly two locations: the module's own file and `COMMON/CMakeLists.txt:5`. None of its public symbols (`root_solver_talk`, `fzero1d_bisec`, `fzero1d_newton`, `f2c_wrapper_*`, `test_func1d_param1`, `test_fdfunc1d_param1`, declared at lines 57-75) appear outside the file.
- One near-miss to be aware of during review: `COMMON/gsl_integration_routines_mod.f90` also contains a `f2c_wrapper_func1d_param1`, but it is a module-local `CONTAINS` definition at line 254 and that file does not `use gsl_roots_routines_mod`. The name overlap is coincidental, not a dependency.

Work against #88 is branched but unmerged. Confirmed commits, each an ancestor of the named branch (branch tips have moved since):

| branch | commit | subject |
|---|---|---|
| `origin/reorder/p2-besselk` (tip `dbbacf2`) | `a58692e` | besselk: back gsl_specialfunctions_mod by fortnum bessel_kn |
| `origin/reorder/p3-quadpack` (tip `6a80e32`) | `76de487` | quadpack: back gsl_integration_routines_mod by fortnum, drop unused gsl_roots |
| `origin/reorder/p4-bspline` (tip `1942289`) | `a24df49` | bspline: back gsl_bspline_routines_mod by fortnum_bspline |
| `origin/reorder/p5-drop-gsl` (tip `e93c2f6`) | `dfdd5b6` | cmake: drop GSL and FGSL now that fortnum backs every module |

`git branch -r --merged origin/main` lists none of them, and `main` still contains all four wrappers. `76de487` independently reaches the same conclusion about the roots module — its message states "gsl_roots_routines_mod has no call sites in the tree and is removed together with its build-list entry" — but it bundles that deletion with the quadrature port, which is gated on a fortnum milestone. Splitting the deletion out is what makes it landable now.

## Reproduction / evidence

```
$ cd NEO-2 && git rev-parse --short HEAD
80a6032
$ rg -n 'gsl_roots' .
./COMMON/gsl_roots_routines_mod.f90:2:! module: gsl_roots_routines_mod ...
./COMMON/gsl_roots_routines_mod.f90:14:MODULE gsl_roots_routines_mod
./COMMON/gsl_roots_routines_mod.f90:349:END MODULE gsl_roots_routines_mod
./COMMON/CMakeLists.txt:5:      gsl_roots_routines_mod.f90
```

A 349-line module that `use fgsl` and is compiled on every build, with zero consumers.

## Behaviour (BDD)

```gherkin
Feature: FGSL surface reduction

  Scenario: dead root-finding wrapper is gone
    Given a checkout of NEO-2 after this change
    When I grep the tree for "gsl_roots"
    Then there are no matches in source or build files

  Scenario: the build is unaffected
    Given the same checkout
    When I run "make" and then "make test"
    Then configuration, compilation and ctest all succeed
    And libcommon no longer contains objects from gsl_roots_routines_mod

  Scenario: remaining wrappers are untouched
    Given the same checkout
    When I inspect COMMON/CMakeLists.txt
    Then gsl_specialfunctions_mod.f90, gsl_integration_routines_mod.f90 and
         gsl_bspline_routines_mod.f90 are still built
    And FGSL and GSL::gsl are still linked
```

## Implementation plan

1. Delete `COMMON/gsl_roots_routines_mod.f90`.
2. Remove line 5 (`gsl_roots_routines_mod.f90`) from `FGSL_USERINTERFACE_SOURCE_FILES` in `COMMON/CMakeLists.txt`.
3. Nothing else. Do not touch `cmake/fgsl.cmake`, the `GSL::gsl` link, or the other three wrappers — those belong to #88's leaf table and to the `reorder/p*` stack.

One PR, branched from `main`, body `Closes #18`, per the PR rules in #88.

## Acceptance criteria

- `rg -n 'gsl_roots' .` returns no matches.
- `make` succeeds from a clean build directory.
- `make test` (`ctest --test-dir TEST`) passes with the same result set as on `80a6032`.
- `git diff --stat` touches exactly two paths: the deleted module and `COMMON/CMakeLists.txt`.
- FGSL and GSL are still found and linked; this PR does not claim any licensing change.

## References

- itpplasma/NEO-2#88 — migration tracker. Note that #88's own text says "Issue #18 tracks the technical replacement", i.e. #88 was opened to make the *license* blocker explicit and deferred the technical work to this issue; the 2026-06-10 comments then made #88 the coordination point while explicitly keeping #18 open. #88 also states that leaf issues are created "only once the named fortnum milestone is closed", which is why the unblocked deletion needs a home outside that table.
- itpplasma/MEPHIT#24 — schedules `src/mephit_pert.f90`'s `gsl_sf_bessel_In_array` `bind(C)` interface for replacement by `fortnum_special`; this is why the "see MEPHIT for examples" pattern should not be copied.
- itpplasma/libneo#291 — umbrella.
- No specific literature reference; this is an implementation detail internal to the code.

## Scope and risk

Very low risk. Deleting unreferenced code cannot change numerical results, and the change is mechanically verifiable by grep plus a clean build. The only real risk is a consumer reached through a path the grep missed — a preprocessor include, or a downstream repo linking `libcommon` and `use`-ing the module directly. The grep covers the NEO-2 tree only; if any downstream (SIMPLE, MEPHIT, KAMEL) is known to `use gsl_roots_routines_mod`, say so on the issue and this rescoping is void.

If maintainers prefer to keep #18 purely as historical context and land the deletion under #88 instead, that is fine — but then it should be a named leaf in #88's table with an explicit "not blocked by any fortnum milestone" note, because today it is invisible in that plan.
